src/plugins/embeddable/public/lib/containers/embeddable_child_panel.tsx

```diff
@@ -52,8 +61,42 @@ export class EmbeddableChildPanel extends React.Component<EmbeddableChildPanelPr
     const { container } = this.props;
 
     this.embeddable = await container.untilEmbeddableLoaded(this.props.embeddableId);
+
+    let startTime = 0;
+    this.embeddable
+      .getOutput$()
+      .pipe(
+        /**
+         * Record start time if loading === true
+         * Forward events only if loading === false
+         */
+        filter((output) => {
+          if (output.loading === true) {
+            // Record start time
+            startTime = performance.now();
+          }
+          return output.loading === false;
+        }),
+        // Map loaded event properties
+        map((output) => {
+          const endTime = new Date().getTime();
+          return {
+            id: this.embeddable.id,
+            status: output.error ? 'error' : 'loaded',
+            error: output.error,
+            timeTookMs: endTime - performance.now(),
```

Comment:
Should this be `performance.now() - startTime` (in any case mixing `new Date().getTime()` and `performance.now()` was probably not intentional)?